python/lsst/pipe/base/task.py

```diff
@@ -370,7 +370,30 @@ def _computeFullName(self, name):
         """
         return f"{self._fullName}.{name}"
 
+    @staticmethod
+    def _unpickle_via_factory(factory, args, kwargs):
+        """Unpickle something by calling a factory
+
+        Allows subclasses to unpickle using `__reduce__` with keyword
+        arguments as well as positional arguments.
+        """
+        return factory(*args, **kwargs)
+
+    def _reduce_kwargs(self):
+        """Returns a dict of the keyword arguments that should be used
+        by `__reduce__`.
+
+        Subclasses with additional arguments should always call the parent
+        class method to ensure that the standard parameters are included.
+
+        Returns
+        -------
+        kwargs : `dict`
+            Keyword arguments to be used when pickling.
+        """
+        return dict(config=self.config, name=self._name, parentTask=self._parentTask,)
```

Comment:
Doesn't `PipelineTask` need a `_reduce_kwargs()` override that includes `initInputs`?